devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -31,57 +33,96 @@ public AddExtensions(final ProjectWriter writer, final String pom) throws IOExce
         this.pom = pom;
     }
 
-    public boolean addExtensions(final Set<String> extensions) throws IOException {
+    /**
+     * Selection algorithm.
+     *
+     * @param query the query
+     * @param extensions the extension list
+     * @return the list of matching candidates and whether or not a match has been found.
+     */
+    static SelectionResult select(String query, List<Extension> extensions) {
+        String q = query.trim().toLowerCase();
+
+        // Try exact matches
+        Set<Extension> matchesNameOrArtifactId = extensions.stream().filter(extension -> extension.getName().equalsIgnoreCase(q)
+                || matchesArtifactId(extension.getArtifactId(), q)).collect(Collectors.toSet());
+
+        if (matchesNameOrArtifactId.size() == 1) {
+            return new SelectionResult(matchesNameOrArtifactId, true);
+        }
+
+        // Try short names
+        Set<Extension> matchesShortName = extensions.stream().filter(extension -> matchesShortName(extension, q))
+                .collect(Collectors.toSet());
+
+        if (matchesShortName.size() == 1) {
+            return new SelectionResult(matchesShortName, true);
+        }
+
+        // find by labels
+        List<Extension> matchesLabels = extensions.stream()
+                .filter(extension -> extension.labels().contains(q)).collect(Collectors.toList());
+
+        Set<Extension> candidates = new LinkedHashSet<>();
+        candidates.addAll(matchesNameOrArtifactId);
+        candidates.addAll(matchesShortName);
+        candidates.addAll(matchesLabels);
+        return new SelectionResult(candidates, false);
+    }
+
+    private static boolean matchesShortName(Extension extension, String q) {
+        return q.equalsIgnoreCase(extension.getShortName());
+    }
+
+    private static boolean matchesArtifactId(String artifactId, String q) {
+        return (artifactId.equalsIgnoreCase(q) ||
+                artifactId.equalsIgnoreCase("quarkus-" + q) ||
+                artifactId.equalsIgnoreCase("quarkus-smallrye-" + q));
```

Comment:
Is there consensus on that one. I through the idea but was not 100% sure myself.

Reply:
For example the `rest-client`, we are discuss which REST Client we really want. What if we don't go for the smallrye one for the selected one?

Reply:
That would mean we would have 2 rest client? 

Where this is useful is to include things like `metrics` or `health` using these simple words. Now we can have the same behavior using `shortName` for these. 

@kenfinnigan WDYT?

Reply:
Currently we have the SmallRye REST Client, once there's a RESTEasy 4.x release with the MP REST Client impl there we would remove SmallRye REST Client and only go with RESTEasy one

Reply:
So do we keep this match or not?

If we drop it, we may want to add short names to the different smallrye implementation (metrics, health...).

Reply:
(Forgot to say, no strong opinion)

Reply:
I can see it makes sense, though it saddens me that SmallRye becomes hidden completely in Quarkus.

Reply:
The groupIds of the dependency are still visible (that does not change). It's only how the user selects the extension.

1. Smallrye is still visible in list-extension.
2. The user can do `add-extension -Dextension=smallrye-health` (he can even use the full artifactId)
3. The only difference is: the user can also do: `add-extension -Dextension=health` which would print on screen: smallrye-health has been added (or something like this, but the extension name is used in this case).

Reply:
Right my point is that if at some point we don’t want the SR version of something then we would be stuck. I prefer that we use short names then for the ones we want to shorten and recommend. Instead of that rule that is.

Reply:
Ok, will update.

Reply:
Removed and short names added.